src/main/java/com/mongodb/jdbc/ExtJsonValue.java

```diff
@@ -0,0 +1,109 @@
+package com.mongodb.jdbc;
+
+import static org.bson.codecs.configuration.CodecRegistries.fromProviders;
+
+import java.io.StringWriter;
+import java.sql.Date;
+import java.text.SimpleDateFormat;
+import org.bson.BsonArray;
+import org.bson.BsonValue;
+import org.bson.codecs.BsonValueCodecProvider;
+import org.bson.codecs.Codec;
+import org.bson.codecs.EncoderContext;
+import org.bson.codecs.configuration.CodecRegistry;
+import org.bson.json.JsonMode;
+import org.bson.json.JsonWriterSettings;
+
+/**
+ * ExtJsonValue is a wrapper for BsonValue. The purpose of this class is to override the toString()
+ * method to produce the extended JSON representation of a BsonValue, rather than the java driver's
+ * default BsonValue.toString() output.
+ *
+ * <p>The driver's BsonValue class is abstract and intentionally cannot be extended by third
+ * parties. The driver explains this is to keep the BSON type system closed.
+ */
+public class ExtJsonValue {
```

Comment:
One drive-by request from me: could we make this something more generically named than `ExtJsonValue` (perhaps something like `MongoSQLValue`)? If we ever want to do something in the future like serialize these strings differently or make string serialization configurable, this class name might become outdated (but we won't be able to change it without it _technically_ being backwards breaking)

Reply:
sure thing. I wasn't really sure what name would be good here. I'll make this change today.

Reply:
The proposed name on the design doc was "MongoSQLBsonValue" but "MongSQLValue" is just as good.

Reply:
done, but could still change this to `MongoSQLBsonValue`. Sorry I missed that in the design 😓

Reply:
Updated to be consistent with design doc.